### PR TITLE
implement sport detail page with api connection

### PR DESCRIPTION
**Pull request by Supakrit65**
fsdffdsdfs

- Connect to backend api
- Try replicating mockup layout
- Change relative path to domain root

### CODE REVIEW

File: src/pages/sportDetail.tsx [hunk 1]
```diff
@@ -1,83 +1,146 @@
...
+    }, [id]);
 
     const goToNextPage = () => {
-        const nextPage = parseInt(id, 10) + 1;
+        const nextPage = parseInt(id!, 10) + 1;
+        navigate(`/sports/${nextPage}`);
+    };
+
+    const goToPreviousPage = () => {
+        const nextPage = parseInt(id!, 10) - 1;
```
Comment (Savetang19) [inline src/pages/sportDetail.tsx]: Please change const name from  `nextPage` to `previousPage`

File: src/pages/sportDetail.tsx
Comment (Savetang19) [inline src/pages/sportDetail.tsx:1]: - Can you handle the error that for `any sport that doesn't exist`, change it to redirect to `sport ID 1`.
- For path `/sports/0` it shows `Archery` which its ID is `1`.

**Review by Savetang19 — CHANGES_REQUESTED** [review]

File: src/pages/sportDetail.tsx [hunk 2]
```diff
@@ -1,83 +1,146 @@
...
+        } catch (error) {
+            console.error("Error fetching sport data: ", error);
+            navigate(`/sports/1`);
+        } finally {
+            setLoading(false);
+        }
+    };
+
+    useEffect(() => {
+        if (Number(id) > 0 && id) { fetchSport(id); }
```
Comment (GToidZ) [inline src/pages/sportDetail.tsx]:
```suggestion
        if (id && parseInt(id, 10) > 0) { fetchSport(id); }
```

Please check for null before parsing integer.

**Review by GToidZ — CHANGES_REQUESTED** [review]
A little bit more change won't hurt 😀

**Review by Savetang19 — APPROVED** [review]
image

LGTM!!! Nice ๆ

**Review by RyukungG — APPROVED** [review]
Good ๆ

**Review by GToidZ — APPROVED** [review]
LGTM

**Pull request merged**